File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int arr[] = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        for(int i=0; i<N; i++) {
        	arr[i] = Integer.parseInt(st.nextToken());
        }
        int cmp = Integer.parseInt(br.readLine());
        int cnt = 0;
        int left = 0;
        int right = arr.length-1;
        int answer = 0;
        Arrays.sort(arr);
        while(left <  right) {
        	answer = arr[left] + arr[right];
        	if(answer == cmp) {
        		cnt ++;
        	}
        	if(answer <= cmp) {
        		left ++;
        	}
        	else {
        		right --;
        	}
        }
        System.out.println(cnt);
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
#include<iostream>
using namespace std;


int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n, x;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        int tmp;
        cin >> tmp;
        arr[i] = tmp;
    }
    sort(arr.begin(), arr.end());
    cin >> x;
    int cnt = 0;
    int left = 0;
    int right = n - 1;
    while (left < right) {
        int tmp = arr[left] + arr[right];
        if (tmp == x) {
            cnt++;
        }
        if (tmp <= x) {
            left++;
        }
        else {
            right--;
        }
    }
    cout << cnt << endl;
}